pkg/security/probe/serializers.go

```diff
@@ -570,7 +570,7 @@ func newUnloadModuleEventSerializer(e *Event) *ModuleEventSerializer {
 
 func serializeSyscallRetval(retval int64) string {
 	switch {
-	case syscall.Errno(retval) == syscall.EACCES || syscall.Errno(retval) == syscall.EPERM:
+	case int(syscall.Errno(retval)) == -int(syscall.EACCES) || int(syscall.Errno(retval)) == -int(syscall.EPERM):
```

Comment:
I don't understand the issue here, here is how golang does it https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/syscall/syscall_unix.go;l=140

Reply:
Maybe we should do `syscall.Errno(-retval)` ?

Reply:
It's a signing issue. In kernel space we retrieve `-EPERM` but here we check `EPERM` ... and `syscall.Errno` is a `uintptr` so you can't use negative values ...

Reply:
but yes, if we move it down under `retval < 0` we could do what you proposed 👍🏻

Reply:
haha great minds think alike :)

Reply:
yes I like it a lot more, thank you !